### ISSUE TITLE
Latest changes show index bounds exception

### ISSUE THREAD

**Issue by mubaldino** [OpenSextant/Gazetteer#2]
Index out of bounds (index 17 of size 17) exception occurs about 5 times during the final build of the Merged.   I presume this is some change that is looking for the 18th field (0..17) in an array of length 17.

symptom -- logged exceptions, and the "MergedGazetteer.txt" is not produced.
Easy enough to reproduce on small gaz input.

INFO  02-08 17:19:13,905 - BuildMergedGazetteer - Starting entry [Estimate Place biases]
INFO  02-08 17:19:13,905 - Estimate Place biases - Loading transformation from XML file [file:///data/build/OpenSextant/Gazetteer/GazetteerETL/EstimateBiases.ktr]
ERROR 02-08 17:19:14,042 - BuildMergedGazetteer - org.pentaho.di.core.exception.KettleException:
Unexpected error during transformation metadata load
Index: 17, Size: 17

```
    at org.pentaho.di.job.entries.trans.JobEntryTrans.getTransMeta(JobEntryTrans.java:1202)
    at org.pentaho.di.job.entries.trans.JobEntryTrans.execute(JobEntryTrans.java:621)
    at org.pentaho.di.job.Job.execute(Job.java:528)
    at org.pentaho.di.job.Job.execute(Job.java:667)
    at org.pentaho.di.job.Job.execute(Job.java:667)
    at org.pentaho.di.job.Job.execute(Job.java:667)
    at org.pentaho.di.job.Job.execute(Job.java:667)
    at org.pentaho.di.job.Job.execute(Job.java:667)
    at org.pentaho.di.job.Job.execute(Job.java:393)
    at org.pentaho.di.job.Job.run(Job.java:313)
```

Caused by: java.lang.IndexOutOfBoundsException: Index: 17, Size: 17
        at java.util.ArrayList.rangeCheck(ArrayList.java:653)
        at java.util.ArrayList.get(ArrayList.java:429)
        at org.pentaho.di.core.logging.TransLogTable.loadXML(TransLogTable.java:143)
        at org.pentaho.di.trans.TransMeta.loadXML(TransMeta.java:2663)
        at org.pentaho.di.trans.TransMeta.<init>(TransMeta.java:2376)
        at org.pentaho.di.trans.TransMeta.<init>(TransMeta.java:2337)
        at org.pentaho.di.job.entries.trans.JobEntryTrans.getTransMeta(JobEntryTrans.java:1158)
        ... 9 more

ERROR 02-08 17:19:14,043 - BuildMergedGazetteer - org.pentaho.di.core.exception.KettleException:
Unexpected error occurred while launching entry [Estimate Place biases.0]

**Comment by mubaldino** [comment]
the update that fixed the missing Lee county did not impact this bug.

**Comment by dlutz2** [comment]
Marc;
 This error seems to show that the Estimate Place Biases transformation cannot start for when the BuildMergedGazetteer jon tries to start it..
Not sure why it would happen more than once in a session.
Do you see this error when you run the Estimate Place Biases transformation by itself in Spoon?

**Comment by dlutz2** [comment]
Marc;
 Haven't been able to reproduce this either with full job or Estimate Place Biases transformation by itself.
Could this be a Mac vs Windows thing?
Some of the forums report this as an intermittent error but no clear solution. Usually involves something to do with the "repo" which in our case is just the directory in which the Kettle scripts reside.
 Zombie Java process from earlier failed run holding onto transformation script?

**Comment by mubaldino** [comment]
Moving to Kettle 6.1.   You note deficiencies with 6.0 and 5.0...
With 3.x I was locked into JDK 6.  Time to upgrade.

I was running all this from the one BuildMerged.. kjb script.
I'm not sure why it reports exceptions 5 times;  Once per data set or resource?
I will try to repoduce on a clean checkout on Mac.
The above trials were on Linux.

Thanks for looking at it.

marc

**Comment by dlutz2** [comment]
Marc;
I just tried with Kettle 6.1 and it looks good: I don't see either the .xls file issue or the Java issue.
I'll triy a few more test and update the README

**Comment by mubaldino** [comment]
Just about confirmed.
An aside:  with Kettle 6.1 the logging warnings appear to be interferring with the ETL operation.

Kettle app log message here is repeated 100s of times.  I see very little OpenSextant output from the Kettle job; shall I infer "—minimal" cmd line option is in effect?

cfgbuilder - Warning: The configuration parameter [org] is not supported by the default configuration builder for scheme: sftp

thanks,
m

**Comment by dlutz2** [comment]
Marc;
That cfgbuilder warning is supposed to be harmless (aside from filling up the logs) if inscrutable according to the JIRA (to be fixed in 7.0). There doesn't seem to be any way to turn it off without turning off most logging. The current script does set the logging level to minimal.

**Comment by mubaldino** [comment]
Thanks.  Confirmed Kettle upgrade fixed this issue.
